volatility/framework/plugins/windows/netscan.py

```diff
@@ -0,0 +1,306 @@
+                # 8.1
+                filename = "netscan-win81-{arch}".format(arch=arch)
+        else:
+            # default to a fallback, but this *should* not happen.
+            filename = "netscan-win{vers}-{arch}".format(vers=major_version, arch=arch)
+
+        vollog.debug("Determined symbol filename: {}".format(filename))
+
+        return filename, class_types
+
+    def create_netscan_symbol_table(self) -> str:
+        """Creates a symbol table for TCP Listeners and TCP/UDP Endpoints.
+
+        Returns:
+            The name of the constructed symbol table
+        """
+        table_mapping = {"nt_symbols": self.config["nt_symbols"]}
+
+        symbol_filename, class_types = self.determine_tcpip_version()
+
+        return intermed.IntermediateSymbolTable.create(self.context,
+                                                       self.config_path,
+                                                       "windows",
+                                                       symbol_filename,
+                                                       class_types = class_types,
+                                                       table_mapping = table_mapping)
+
+    @classmethod
+    def scan(cls,
+             context: interfaces.context.ContextInterface,
+             layer_name: str,
+             nt_symbol_table: str,
+             netscan_symbol_table: str) -> \
+        Iterable[interfaces.objects.ObjectInterface]:
+        """Scans for network objects using the poolscanner module and constraints.
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            layer_name: The name of the layer on which to operate
+            nt_symbol_table: The name of the table containing the kernel symbols
+            netscan_symbol_table: The name of the table containing the network object symbols (_TCP_LISTENER etc.)
+
+        Returns:
+            A list of network objects found by scanning the `layer_name` layer for network pool signatures
+        """
+
+        constraints = cls.create_netscan_constraints(context, netscan_symbol_table)
+
+        for result in poolscanner.PoolScanner.generate_pool_scan(context, layer_name, nt_symbol_table, constraints):
+
+            _constraint, mem_object, _header = result
+            yield mem_object
+
+    def _generator(self, show_corrupt_results = None):
+        """ Generates the network objects for use in rendering. """
+
+        netscan_symbol_table = self.create_netscan_symbol_table()
+
+        for netw_obj in self.scan(self.context, self.config['primary'], self.config['nt_symbols'],
+                                  netscan_symbol_table):
+
+            vollog.debug("Found netw obj @ 0x{:2x} of assumed type {}".format(netw_obj.vol.offset, type(netw_obj)))
+            # objects passed pool header constraints. check for additional constraints if strict flag is set.
+            if not show_corrupt_results:
```

Comment:
It feels like this could be an and statement, but I'm not too worried about it.  5:)

Reply:
updated.